cellprofiler/modules/activecontourmodel.py

```
@@ -0,0 +1,323 @@
+# coding=utf-8
+
+"""
+
+Active contour model
+
+"""
+
+import cellprofiler.module
+import cellprofiler.object
+import cellprofiler.setting
+import numpy
+import scipy.ndimage
+import skimage.draw
+import skimage.morphology
+import skimage.filters
+import skimage.measure
+import skimage.segmentation
+
+
+class ActiveContourModel(cellprofiler.module.ImageSegmentation):
+    module_name = "Active contour model"
+
+    variable_revision_number = 1
+
+    def create_settings(self):
+        super(ActiveContourModel, self).create_settings()
+
+        self.iterations = cellprofiler.setting.Integer(
+            text="Iterations",
+            value=20
+        )
+
+        self.alpha = cellprofiler.setting.Float(
+            text="Alpha",
+            value=0.2
+        )
+
+        self.threshold = cellprofiler.setting.Float(
+            text="Threshold",
+            value=0
+        )
+
+    def settings(self):
+        __settings__ = super(ActiveContourModel, self).settings()
+
+        return __settings__ + [
+            self.iterations,
+            self.alpha,
+            self.threshold
+        ]
+
+    def visible_settings(self):
+        __settings__ = super(ActiveContourModel, self).settings()
+
+        return __settings__ + [
+            self.iterations,
+            self.alpha,
+            self.threshold
+        ]
+
+    def run(self, workspace):
+        x_name = self.x_name.value
+
+        y_name = self.y_name.value
+
+        images = workspace.image_set
+
+        x = images.get_image(x_name)
+
+        x_data = x.pixel_data
```

Should we raise an error if the image is 2D?